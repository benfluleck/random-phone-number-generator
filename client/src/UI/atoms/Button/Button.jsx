import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import setClassNames from '../../../helpers/utils/setClassNames';

const Button = ({ children, isDisabled, onClick, left, right }) => (
  <Button.Container
    className={setClassNames({
      // eslint-disable-next-line quote-props
      'button': true,
      'is-large': true,
      'is-link': true,
      'is-outlined': true,
      'is-fullwidth': true,
      'is-pulled-left' : left,
      'is-pulled-right' : right,
    })}
    disabled={isDisabled && 'disabled'}
    onClick={onClick}
  >
    {children}
  </Button.Container>
);

Button.Container = styled.button``;

Button.defaultProps = {
  isDisabled: false,
  right: false,
  left: false,
  onClick: () => { },
};

Button.propTypes = {
  children: PropTypes.oneOfType([ PropTypes.string, PropTypes.node ]).isRequired,
  isDisabled: PropTypes.bool,
  onClick: PropTypes.func,
  left: PropTypes.bool,
  right: PropTypes.bool,
};

export default Button;

