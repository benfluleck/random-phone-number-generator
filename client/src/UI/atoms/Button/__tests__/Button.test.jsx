import React from 'react';

import { shallow } from 'enzyme';
import Button from '../Button';

const setup = (props) => {
  const defaultProps = {
    onClick: () => { },
  };
  const wrapper = shallow(
    <Button {...props} {...defaultProps}>
      Test button
    </Button>
  );

  return {
    wrapper,
  };
};


describe('Component - Button', () => {
  test('should render', () => {
    const { wrapper } = setup();

    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
