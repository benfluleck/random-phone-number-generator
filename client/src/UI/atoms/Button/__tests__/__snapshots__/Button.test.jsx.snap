// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Button should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    isDisabled={false}
    left={false}
    onClick={[Function]}
    right={false}
  >
    Test button
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "Test button",
      "className": "button is-large is-link is-outlined is-fullwidth",
      "disabled": false,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "Test button",
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [],
      },
      "displayName": "styled.button",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "button",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "Test button",
        "className": "button is-large is-link is-outlined is-fullwidth",
        "disabled": false,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Test button",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [],
        },
        "displayName": "styled.button",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "button",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
